File: samples/android-app/src/main/kotlin/io/github/andrewmalitchuk/uncial/samples/android/ui/source/kit/molecule/item/BlockRow.kt
package io.github.andrewmalitchuk.uncial.samples.android.ui.source.kit.molecule.item

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import io.github.andrewmalitchuk.uncial.samples.android.ui.core.Theme
import io.github.andrewmalitchuk.uncial.samples.android.ui.source.kit.atom.surface.AppCard

/**
 * One reconstructed block, labelled with what `DocumentStructure` decided it was.
 *
 * The `H1` / `¶` glyph is the point of the screen: the SDK returns structure, and a sample
 * that rendered the blocks as undifferentiated paragraphs would be hiding its own result.
 * The text itself is rendered in the serif — it is the document speaking, not the app.
 */
@Composable
fun BlockRow(
    glyph: String,
    text: String,
    meta: String,
    modifier: Modifier = Modifier,
    isHeading: Boolean = false,
) {
    AppCard(modifier = modifier) {
        Row(verticalAlignment = Alignment.Top) {
            Box(
                modifier = Modifier
                    .size(28.dp)
                    .clip(Theme.corner.small)
                    .background(if (isHeading) Theme.color.selected else Theme.color.surfaceVariant),
                contentAlignment = Alignment.Center,
            ) {
                Text(text = glyph, style = Theme.typography.mono, color = Theme.color.inkMuted)
            }
            Column(
                modifier = Modifier
                    .weight(1f)
                    .padding(start = Theme.spacing.row),
                verticalArrangement = Arrangement.spacedBy(Theme.spacing.tight),
            ) {
                Text(
                    text = text,
                    style = Theme.typography.recognized,
                    fontWeight = if (isHeading) FontWeight.SemiBold else FontWeight.Normal,
                    color = Theme.color.ink,
                )
                Text(text = meta, style = Theme.typography.mono, color = Theme.color.inkSubtle)
            }
        }
    }
}

File: samples/ios-app/UncialSample/Kit/BlockRow.swift
import SwiftUI

/// One reconstructed block, labelled with what `DocumentStructure` decided it was.
///
/// The `H1` / `¶` glyph is the point of the screen: the SDK returns structure, and a
/// sample that rendered the blocks as undifferentiated paragraphs would be hiding its own
/// result. The text itself is in the serif — it is the document speaking, not the app.
struct BlockRow: View {

    let glyph: String
    let text: String
    let meta: String
    var isHeading: Bool = false

    var body: some View {
        AppCard {
            HStack(alignment: .top, spacing: Theme.spacing.row) {
                Text(glyph)
                    .font(Theme.typography.mono)
                    .foregroundColor(Theme.color.inkMuted)
                    .frame(width: 28, height: 28)
                    .background(isHeading ? Theme.color.selected : Theme.color.surfaceVariant)
                    .clipShape(
                        RoundedRectangle(cornerRadius: Theme.corner.small, style: .continuous)
                    )
                VStack(alignment: .leading, spacing: Theme.spacing.tight) {
                    Text(text)
                        .font(Theme.typography.recognized)
                        .fontWeight(isHeading ? .semibold : .regular)
                        .foregroundColor(Theme.color.ink)
                        .textSelection(.enabled)
                    Text(meta)
                        .font(Theme.typography.mono)
                        .foregroundColor(Theme.color.inkSubtle)
                }
            }
        }
    }
}
